packages/reports/src/Components/Errors/error-component.tsx

```diff
@@ -1,23 +1,34 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import { PageError, Dialog } from '@deriv/components';
 import { routes } from '@deriv/shared';
 import { localize } from '@deriv/translations';
 
+type TErrorComponent = {
+    header?: string;
+    is_dialog?: boolean;
+    message?: React.ReactElement | string;
+    redirect_label?: string;
+    redirectOnClick?: () => void;
+    should_show_refresh?: boolean;
+    type?: string;
+};
+
 const ErrorComponent = ({
     header,
     message,
     is_dialog,
     redirect_label,
     redirectOnClick,
     should_show_refresh = true,
-}) => {
+}: TErrorComponent) => {
```

Comment:
instead of defining `?` to all the type rules you can use Partial to replace the `?` 

```suggestion
}: Partial<TErrorComponent>) => {
```